File: Submissions/#0101_div2/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    String S = sc.next();
    String T = sc.next();
    String R = sc.next();
    sc.close();

    char[] s = S.toCharArray();
    char[] t = T.toCharArray();
    char[] r = R.toCharArray();

    int[] countA = new int[26];

    for (int i = 0; i < s.length; i++) {
      int v =(int) s[i] - 65;
      countA[v]++;
    }
  
    for (int i = 0; i < t.length; i++) {
      int v =(int) t[i] - 65;
      countA[v]++;
    }

    int[] countB = new int[26];

    for (int i = 0; i < r.length; i++) {
      int v =(int) r[i] - 65;
      countB[v]++;
    }

    String answer = "YES";

    for (int i = 0; i < 26; i++) {
      if (countA[i] != countB[i]){
        answer = "NO";
        break;
      }
    }

    System.out.println(answer);

  }

}

File: Submissions/#0101_div2/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  string s, t, r;
  cin >> s >> t >> r;

  string answer = "YES";

  vector<int> countA(26);

  for (int i = 0; i < s.size(); i++) {
    int v =(int) s[i] - 65;
    countA[v]++;
  }

  for (int i = 0; i < t.size(); i++) {
    int v =(int) t[i] - 65;
    countA[v]++;
  }

  vector<int> countB(26);

  for (int i = 0; i < r.size(); i++) {
    int v =(int) r[i] - 65;
    countB[v]++;
  }

  for (int i = 0; i < 26; i++) {
    if (countA[i] != countB[i]){
      answer = "NO";
      break;
    }
  }

  cout << answer << endl;

}